iravelmakina/interpreter
# Python test script

### Description:
- Implemented a test suite for the interpreter program
- Added various test cases to cover different input scenarios

### Changes:
- Created `run_program` function to execute the interpreter program
- Developed `test_program` function to run all the test cases
- Included test cases for arithmetic operations, syntax errors, and function usage

### How to Test:
1. Clone the repository
2. Ensure `interpreter.exe` is in the working directory
3. Run the test script with `python test_script.py`
4. Verify the output matches the expected results for all test cases

### Screenshots:
![image](https://github.com/user-attachments/assets/17ec8da2-dd37-47bb-812a-aee8f3e9fe7e)
![image](https://github.com/user-attachments/assets/2b3ff4a3-8b8c-4b9a-956c-86aaad72cec4)
![image](https://github.com/user-attachments/assets/cfadb34b-f0ee-4514-880c-23d7c62cfd3e)
![image](https://github.com/user-attachments/assets/291cb4f0-d2ca-4adc-b41a-c626ae6ce304)
